manager/integration/tests/test_rwx.py

```diff
@@ -572,4 +575,78 @@ def test_rwx_offline_expansion(): # NOQA
+    pvc['spec']['accessModes'] = ['ReadWriteMany']
+    pvc['spec']['resources']['requests']['storage'] = str(1 * Gi)
+
+    core_api.create_namespaced_persistent_volume_claim(
+        body=pvc, namespace='default')
+
+    deployment = make_deployment_with_pvc(
+        'deployment-rwx-expand-test', pvc_name, replicas=1)
+    apps_api = get_apps_api_client()
+    create_and_wait_deployment(apps_api, deployment)
+
+    pv_name = get_volume_name(core_api, pvc_name)
+    share_manager_name = 'share-manager-' + pv_name
+    deployment_label_selector = "name=" + \
+                                deployment["metadata"]["labels"]["name"]
+
+    deployment_pod_list = \
+        core_api.list_namespaced_pod(namespace="default",
+                                     label_selector=deployment_label_selector)
+
+    assert deployment_pod_list.items.__len__() == 1
+
+    pod_name = deployment_pod_list.items[0].metadata.name
+    test_data = generate_random_data(VOLUME_RWTEST_SIZE)
+    write_pod_volume_data(core_api, pod_name, test_data, filename='test')
+
+    deployment['spec']['replicas'] = 0
+    apps_api.patch_namespaced_deployment(body=deployment,
+                                         namespace='default',
+                                         name=deployment["metadata"]["name"])
+
+    wait_for_volume_detached(client, pv_name)
+    expand_and_wait_for_pvc(core_api, pvc, EXPANDED_VOLUME_SIZE*Gi)
+    wait_for_volume_expansion(client, pv_name)
+    wait_for_volume_detached(client, pv_name)
+
+    deployment['spec']['replicas'] = 1
+    apps_api.patch_namespaced_deployment(body=deployment,
+                                         namespace='default',
+                                         name=deployment["metadata"]["name"])
+    wait_deployment_replica_ready(apps_api, deployment["metadata"]["name"], 1)
+    wait_for_volume_healthy(client, pv_name)
+
+    found = False
+    for i in range(RETRY_COUNTS):
+        pods = core_api.list_namespaced_pod(namespace=LONGHORN_NAMESPACE)
+        pod_names = []
+        for item in pods.items:
+            pod_names.append(item.metadata.name)
+        if share_manager_name in pod_names:
+            found = True
+            break
+        time.sleep(RETRY_INTERVAL)
+    assert found
+
+    deployment_pod_list = \
+        core_api.list_namespaced_pod(namespace="default",
+                                     label_selector=deployment_label_selector)
+    pod_name = deployment_pod_list.items[0].metadata.name
+    data_size_in_mb = 1536
+    write_pod_volume_random_data(core_api, pod_name,
+                                 "/data/test2",
+                                 data_size_in_mb)
+
+    for i in range(RETRY_EXEC_COUNTS):
+        volume = client.by_id_volume(pv_name)
+        actual_size = int(volume.controllers[0].actualSize)
+        actual_size_in_mb = actual_size/1024/1024
+        if actual_size_in_mb > data_size_in_mb:
+            break
+        time.sleep(RETRY_EXEC_INTERVAL)
```

Comment:
Why need retry? to ensure the status of update?

Reply:
Yes I found in SLES the actual size status change will slow and make test case flaky.
But I removed this part in the updated PR code, thank you